File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/utils/WriterUtils.java
package com.raizlabs.android.dbflow.processor.utils;

import com.raizlabs.android.dbflow.processor.definition.BaseDefinition;
import com.raizlabs.android.dbflow.processor.model.ProcessorManager;
import com.squareup.javapoet.JavaFile;

import java.io.IOException;

/**
 * Description: Provides some handy writing methods.
 */
public class WriterUtils {

    public static boolean writeBaseDefinition(BaseDefinition baseDefinition, ProcessorManager processorManager) {
        boolean success = false;
        try {
            JavaFile javaFile = JavaFile.builder(baseDefinition.packageName, baseDefinition.getTypeSpec()).build();
            javaFile.writeTo(processorManager.getProcessingEnvironment().getFiler());
            success = true;
        } catch (IOException e) {
            e.printStackTrace();
        }
        return success;
    }

}

File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/utils/WriterUtils.kt
package com.raizlabs.android.dbflow.processor.utils

import com.grosner.kpoet.javaFile
import com.raizlabs.android.dbflow.processor.ProcessorManager
import com.raizlabs.android.dbflow.processor.definition.BaseDefinition
import java.io.IOException

/**
 * Description: Provides some handy writing methods.
 */
object WriterUtils {

    fun writeBaseDefinition(baseDefinition: BaseDefinition, processorManager: ProcessorManager): Boolean {
        var success = false
        try {
            javaFile(baseDefinition.packageName) { baseDefinition.typeSpec }
                    .writeTo(processorManager.processingEnvironment.filer)
            success = true
        } catch (e: IOException) {
            // ignored
        } catch (i: IllegalStateException) {
            processorManager.logError(WriterUtils::class, "Found error for class:" + baseDefinition.elementName)
            processorManager.logError(WriterUtils::class, i.message)
        }

        return success
    }

}
